src/parser.js

```diff
@@ -1539,6 +1573,12 @@ export class Parser extends Tokenizer {
         return this.markLocation(new Shift.ShorthandProperty(new Shift.Identifier(key.value)), startTokenIndex);
       }
     }
+    if (this.match(TokenType.LBRACK)) {
```

Comment:
Maybe we can consider merging this test to the next if statement to avoid the duplicate code?